features/ultils/world.ts

```diff
@@ -36,6 +42,47 @@ export class OurWorld extends World {
       destFilePath: filePath,
     });
   }
+
+  public getCredentialByAppKey(appKey: string): Credential {
+    const credential = this.credentials?.find((c) => c.key === appKey);
+    if (credential === undefined) {
+      throw new Error(`The credential with app key ${appKey} is not found`);
+    }
+
+    if (!credential.appId) {
+      throw new Error(`The credential with app key ${appKey} has no App ID`);
+    }
+
+    return credential;
+  }
+
+  public getAPITokenByAppAndPermission(
+    appKey: string,
+    permissions: Permission[],
+  ): string {
+    if (permissions.length === 0) {
+      throw new Error(`The permissions is empty`);
+    }
+
+    const credential = this.getCredentialByAppKey(appKey);
+    const apiToken = credential.apiTokens.find((row: ApiToken) => {
+      if (row.permissions.length !== permissions.length) {
+        return false;
+      }
+
+      return permissions.every((value) => row.permissions.includes(value));
+    });
```

Comment:
We need to satisfy the following requirements.
- API token can be generated without permission.
- If the input `permissions` is empty, we need to find a token without permissions.
- We need to return the token with exact permission.

```suggestion
    const credential = this.getCredentialByAppKey(appKey);
    const apiToken = credential.apiTokens.find((row: ApiToken) => {
      // In case the token doesn't have any permissions.
      if (permissions.length === 0 && row.permissions.length === 0) {
        return true;
      }

      if (row.permissions.length !== permissions.length) {
        return false;
      }

      return permissions.every((value) => row.permissions.includes(value));
    });
```

Reply:
You can write unit tests if you want.